### PR TITLE
feat(ui): run deps in UI mode if dep projects are checked

### LINKED ISSUE #22045: [Feature] UI mode doesn't respect test dependencies
Running playwright test with a test that have a dependency. For example the below snippet will run the `setup` dependency when running `npx playwright test`. In UI mode I would reasonably expect the same - in that when running a test that has a dependency it would run the dependency first and then run the test. This is currently not the case.

```{
      name: 'loggedIn',
      use: {
        ...devices['Desktop Chrome'],
        storageState: STORAGE_STATE,
      },
      testIgnore: ['**/*login.spec.ts', '**/*.setup.ts'],
      dependencies: ['setup'],
    },```

### PR DESCRIPTION
**Pull request by pavelfeldman**
Fixes https://github.com/microsoft/playwright/issues/22045

### CODE REVIEW

File: utils/limits.sh
Comment (dgozman) [inline utils/limits.sh:1]: Intended change?

**Review by dgozman — APPROVED** [review]

File: utils/limits.sh
Comment (pavelfeldman) [reply utils/limits.sh:1]: Very much so

**Comment by amitasil** [comment]
Thanks!!

**Comment by mikelhamer** [comment]
Yay!!! Thank you!

**Comment by lattebread** [comment]
Thank you!

**Comment by sarah-gelt** [comment]
Fantastic, thank you!

**Comment by beig** [comment]
where can I find the option?
or what does 'dep projects are checked' mean?

currently my setup dependency is not run in UI mode

**Comment by sarah-gelt** [comment]
> where can I find the option? or what does 'dep projects are checked' mean?
> 
> currently my setup dependency is not run in UI mode

Click in the area highlighted to expand the Projects list. Ensure your dependency projects are checked as well as the projects the tests are in. When you run a test, UI mode will then run the dependencies as well. Hope that helps!
image

**Comment by scamden** [comment]
Thanks for this! One small follow up if you have time would be that if I specify `--project chromium` on the cli, ideally the dependent projects and teardowns would also be checked when I enter UI mode

 (this would mirror the non UI mode functionality of automatically running those projects if I select only the one)

**Comment by scamden** [comment]
Also this line seems to be a little out of date since this was merged:
>  The UI mode will not take into consideration the setup tests and therefore you will have to manually run them first.

https://playwright.dev/docs/test-ui-mode#filtering-tests

**Pull request merged**